youtrack_sdk/types.py

```diff
@@ -0,0 +1,33 @@
+from datetime import datetime
+
+from pydantic.datetime_parse import from_unix_seconds
+from pydantic.errors import PydanticValueError
+
+
+class StrictIntError(PydanticValueError):
+    msg_template = "value is not a valid integer"
+
+
+class IncompatibleFieldTypeError(PydanticValueError):
+    msg_template = "incompatible field type"
+
+
+class DateTime(datetime):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value, values):
+        if (project_custom_field := values.get("project_custom_field")) and (
+            project_custom_field.field.field_type.id != "date and time"
+        ):
+            raise IncompatibleFieldTypeError
+
+        if isinstance(value, datetime):
+            return value
```

Comment:
I wonder under which circumstance this would happen? When you create a class as a user of the SDK and assign a `datetime` to it, correct?

Reply:
Yes, exactly. For example, I can create a new issue with one of the custom fields that looks like this:
```python
SimpleIssueCustomField(
    name="DateTime",
    value=datetime.datetime(2020, 12, 31, 23, 59),
)
```